tests/lib/bin/mountinfo-tool

```diff
@@ -269,12 +279,112 @@ def renumber_snap_revision(entry, seen):
         snap_name = parts[5]
         snap_rev = parts[6]
         compose = compose_alternate
+    elif len(parts) >= 8 and parts[:6] == ["", "var", "lib", "snapd", "hostfs", "snap"]:
+        snap_name = parts[6]
+        snap_rev = parts[7]
+        compose = compose_hostfs_preferred
+    elif len(parts) >= 11 and parts[:9] == [
+        "",
+        "var",
+        "lib",
+        "snapd",
+        "hostfs",
+        "var",
+        "lib",
+        "snapd",
+        "snap",
+    ]:
+        snap_name = parts[9]
+        snap_rev = parts[10]
+        compose = compose_hostfs_alternate
     else:
         return
     n = alloc_n(snap_name, snap_rev)
     entry.mount_point = "/".join(compose(parts, n))
 
 
+class RenumberSnapRevisionTests(unittest.TestCase):
```

Comment:
I would put the tests after the tested code, not before

Reply:
The test class is immediately after the tested function. Did you mean to put it at the end of the whole module?